backend/src/routes/api/notebooks/utils.ts

```diff
@@ -17,29 +15,19 @@ export const getNotebookStatus = async (
   fastify: KubeFastifyInstance,
   namespace: string,
   name: string,
-): Promise<{ notebook: Notebook; isRunning: boolean; podUID: string }> => {
+): Promise<{ notebook: Notebook; isRunning: boolean; podUID: string; notebookLink: string }> => {
   const notebook = await getNotebook(fastify, namespace, name);
   const hasStopAnnotation = !!notebook?.metadata.annotations?.['kubeflow-resource-stopped'];
   const [isRunning, podUID] = hasStopAnnotation
     ? [false, '']
     : await listNotebookStatus(fastify, namespace, name);
-
-  const notebookName = notebook?.metadata.name;
-  let newNotebook: Notebook;
-  if (isRunning && !notebook?.metadata.annotations?.['opendatahub.io/link']) {
-    const route = await getRoute(fastify, namespace, notebookName).catch((e) => {
-      fastify.log.warn(`Failed getting route ${notebookName}: ${e.message}`);
-      return undefined;
-    });
-    if (route) {
-      newNotebook = await patchNotebookRoute(fastify, route, namespace, notebookName).catch((e) => {
-        fastify.log.warn(`Failed patching route to notebook ${notebookName}: ${e.message}`);
-        return notebook;
-      });
-    }
-  }
-
-  return { notebook: newNotebook || notebook, isRunning, podUID };
+  const route = await getRoute(fastify, namespace, name);
+  return {
+    notebook,
+    isRunning,
+    podUID,
+    notebookLink: `https://${route.spec.host}/notebook/${namespace}/${name}`,
+  };
```

Comment:
No case handling if retrieving the route fails? Although I don't think i was able to delete the route to have this fetch fail...

Reply:
Yeah... Maybe only the failed network call can lead to this... let me check, but I believe the operator will always make sure the route is there.